test/sql/test_time_fn/T/test_time_fn

```diff
@@ -197,6 +197,12 @@ select dayofweek_iso("2023-01-03");
 select dayofweek_iso("");
 select dayofweek_iso(NULL);
 
+-- name: test_date_diff
+select date_diff("2010-11-30 23:59:59", "2010-11-30 20:58:59", "millisecond");
+select date_diff("2010-11-30 23:59:59", "2010-11-30 20:58:59", "hour");
+select date_diff("2010-11-30 23:59:59", "2010-11-30 20:58:59", "minute");
+select date_diff("2010-11-30 23:59:59", "2010-11-30 20:59:59", "second");
+
```

Comment:
need more stronger test cases, such as parameters with more precision and some corner cases, such as `SELECT date_diff('2020-03-01 00:00:05.000001','2020-03-02 00:00:05',day');`

Reply:
i would add more testcast for it